Grammar/python.gram

```diff
@@ -142,20 +142,14 @@ while_stmt[stmt_ty]:
     | 'while' a=named_expression ':' b=block c=[else_block] { _Py_While(a, b, c, EXTRA) }
 
 for_stmt[stmt_ty]:
-    | ASYNC 'for' t=star_targets 'in' ex=star_expressions ':' b=block el=[else_block] {
-        _Py_AsyncFor(t, ex, b, el, NULL, EXTRA) }
-    | 'for' t=star_targets 'in' ex=star_expressions ':' b=block el=[else_block] {
-        _Py_For(t, ex, b, el, NULL, EXTRA) }
+    | is_async=[ASYNC] 'for' t=star_targets 'in' ex=star_expressions ':' b=block el=[else_block] {
+        (is_async ? _Py_AsyncFor : _Py_For)(t, ex, b, el, NULL, EXTRA) }
```

Comment:
Wow! Didn't even know that this was possible. Then again, function names are just pointers, so it's logical in hindsight.